Fix: RANSAC warning on small subsets

DavidAGInnovation opened the pull request:
Closes #32908. Prevents RANSAC from scoring subsets with fewer than 2 samples, avoiding UndefinedMetricWarning.

DavidAGInnovation committed:
Fix: RANSAC warning on small subsets (Closes #32908)

DavidAGInnovation committed:
Merge branch 'main' into fix/ransac-warning

DavidAGInnovation committed:
TEST: add regression test for gh-32908

DavidAGInnovation committed:
DOC: add changelog entry for gh-32908

DavidAGInnovation commented:
@lucyleeow

DavidAGInnovation committed:
Merge branch 'main' into fix/ransac-warning

lucyleeow commented:
Hi @DavidAGInnovation , this is better and I will remove the label but we generally want an issue to be triaged (i.e. 'needs triage' label) and have consensus on a solution before anyone makes a PR.
Please read our contributing guide, which will detail etiqutte around contributing.

DavidAGInnovation committed:
Merge branch 'main' into fix/ransac-warning

DavidAGInnovation committed:
Merge branch 'main' into fix/ransac-warning

DavidAGInnovation commented:
I don’t really understand the issue, to be honest. Here you have the code that solves a problem, and to me, it looks like an excuse not to use it.

Feel free to do as you wish, you can also give another excuse to not use/merge the code if you desire, but if we really want to progress, you should definitely focus on the code, not on my activity, nor on other points.

DavidAGInnovation committed:
Merge branch 'main' into fix/ransac-warning

DavidAGInnovation committed:
Merge branch 'main' into fix/ransac-warning

gweberbv-beep commented:
When implementing a fix for this issue, the following should be taken into account: https://github.com/scikit-learn/scikit-learn/issues/32908#issuecomment-3691360744

DavidAGInnovation committed:
Merge branch 'main' into fix/ransac-warning

DavidAGInnovation committed:
Merge branch 'main' into fix/ransac-warning

DavidAGInnovation committed:
FIX RANSAC skip scoring with too few inliers

Avoid calling score on undersized consensus sets (gh-32908) and skip invalid score values.

virchan reviewed (commented):
Thanks for your work on this PR, @DavidAGInnovation.

Since it's been a while without updates and CI is still failing, I'm closing this PR for now. It is also unclear if there's a consensus on the approach to solve #32908.

You're very welcome to reopen if you'd like to continue work on this once there's an agreed direction and when CI is passing.
